### PR TITLE
fix: Fixes for 'ftl init go'

### PR DESCRIPTION
**Pull request by KendallWeihe**
My original goal was to try and `ftl init go` a new module into existence and then deploy it, but the Golang side of the fence hasn't had much attention recently so it didn't quite work.

My intent here wasn't actually to progress FTL forward, but to familiar myself with some of the golang tech. I'm a total newbie to golang, so I'm sure I have some things wrong. If we want to throw this PR aside, then it won't hurt my feelings.

- We were using the `kotlinruntime` in the `ftl go init` command
- I totally deleted `go-runtime/scaffolding/go.mod` because it was being scaffolded into the new module, but we want the template
  - There was [a comment](https://github.com/TBD54566975/ftl/blob/f6bd2854f50ff93e30e97d4e59aa080aaf93230d/go-runtime/scaffolding/go.mod#L1) in there previous which said `// This needs to exist so that the Go toolchain doesn't include this directory. Annoying.` so I'm assuming I can't do this?
- As far as I understand, `latest` isn't a concept in golang? But we had `require github.com/TBD54566975/ftl latest` so I changed that to a concrete version. Again, not sure if we actually want to do this?
- Apparently verb names need to be capitalized? @alecthomas Had to uppercase the `func echo` to `func Echo` because I was getting an error: 

```shell
ftl/dev on  kendallw/ftl-init-go [$✘?] via 🐹 v1.21.5 via ☕ v17.0.6 on 🐳 v24.0.6 took 2.1s
➜ ftl-go deploy alice
info: Building module alice in /Users/kendallw/Library/Caches/ftl-go/build/alice-844bddfb19d1cc96314f8808eb760a45131b6e8e2c3ffc02e66584ce2c5a6c23
info: # main
info: ./main.go:15:18: cannot infer Req (github.com/TBD54566975/ftl@v0.85.3/go-runtime/server/server.go:51:13)
info: ./main.go:15:25: undefined: alice.Echo
ftl-go: error: failed to build module: exit status 1
```

---

### Current State:

We can now do something like...

```shell
(
    ftl init go proof alice
    ftl init go proof bob
    cd proof
    go mod tidy
    ftl-go deploy alice
    ftl-go deploy bob
    echo "Sleeping for deployment..."
    sleep 10
    ftl call alice.echo '{"name": "Alice"}'
    ftl call bob.echo '{"name": "Bob"}'
)
```

**Note:** that `go mod tidy` is needed because our scaffold copies over a empty `go.mod` and no `go.sum` ... maybe we should just include both fully-fleshed-out in the scaffold? 

**Note:** still have to use the `ftl-go` command for golang deployments (among other things, I presume)

### CODE REVIEW

**Comment by alecthomas** [comment]
> There was [a comment](https://github.com/TBD54566975/ftl/blob/f6bd2854f50ff93e30e97d4e59aa080aaf93230d/go-runtime/scaffolding/go.mod#L1) in there previous which said // This needs to exist so that the Go toolchain doesn't include this directory. Annoying. so I'm assuming I can't do this?

Correct. You can leave it there, but add an `scaffolder.Exclude("^go.mod$")` option to `scaffolder.Scaffold(...)`.

File: go-runtime/scaffolding/go.mod.tmpl [hunk 1]
```diff
@@ -1,5 +1,5 @@
 module ftl
 
-go 1.21.0
+go 1.21.5
 
-require github.com/TBD54566975/ftl latest
```
Comment (alecthomas) [inline go-runtime/scaffolding/go.mod.tmpl:5]: This should be fine to leave, as `go mod tidy` should fix it up at build time.

**Review by alecthomas — APPROVED** [review]

**Comment by alecthomas** [comment]
LGTM!

The Go code has been neglected for a bit, thanks for doing some cleanup.

**Comment by alecthomas** [comment]
> As far as I understand, latest isn't a concept in golang? But we had require github.com/TBD54566975/ftl latest so I changed that to a concrete version. Again, not sure if we actually want to do this?

`latest` will be replaced with the latest version when `go mod tidy` is run by the user.

**Comment by KendallWeihe** [comment]
thanks @alecthomas! addressed and merging...

**Pull request merged**